pkg/autoscaling/calculation.go

```diff
@@ -307,35 +324,50 @@ func getScaledTiDBGroups(etcdClient *clientv3.Client, healthyInstances []instanc
 				zap.String("tidb-address", instance.address))
 			return nil
 		}
-		v := tidb.getLabelValue(groupLabelKey)
-		buildPlanMap(planMap, v, instance.address)
+
+		groupName := tidb.getLabelValue(groupLabelKey)
+		if !isAutoScaledGroup(groupName) {
+			continue
+		}
+
+		buildPlanMap(planMap, groupName, instance.address)
+		resourceType := tidb.getLabelValue(resourceTypeLabelKey)
+		if _, ok := resourceTypeMap[groupName]; !ok {
+			if resourceType == "" {
+				log.Warn("tidb is in auto-scaled group but has no resource type label", zap.String("tidb-address", instance.address))
+			} else {
+				resourceTypeMap[groupName] = resourceType
+			}
+		}
 	}
-	return buildPlans(planMap, TiDB)
+	return buildPlans(planMap, resourceTypeMap, TiDB)
+}
+
+func isAutoScaledGroup(groupName string) bool {
+	return len(groupName) > len(autoScalingGroupLabelKeyPrefix) && strings.HasPrefix(groupName, autoScalingGroupLabelKeyPrefix)
 }
 
 func buildPlanMap(planMap map[string]map[string]struct{}, groupName, address string) {
-	if len(groupName) > len(autoScalingGroupLabelKeyPrefix) &&
-		strings.HasPrefix(groupName, autoScalingGroupLabelKeyPrefix) {
-		if component, ok := planMap[groupName]; ok {
-			component[address] = struct{}{}
-		} else {
-			planMap[groupName] = map[string]struct{}{
-				address: {},
-			}
+	if component, ok := planMap[groupName]; ok {
+		component[address] = struct{}{}
+	} else {
+		planMap[groupName] = map[string]struct{}{
+			address: {},
 		}
 	}
 }
 
-func buildPlans(planMap map[string]map[string]struct{}, componentType ComponentType) []*Plan {
+func buildPlans(planMap map[string]map[string]struct{}, resourceTypeMap map[string]string, componentType ComponentType) []*Plan {
 	var plans []*Plan
-	for groupLabel, groupInstances := range planMap {
+	for groupName, groupInstances := range planMap {
 		plans = append(plans, &Plan{
-			Component: componentType.String(),
-			Count:     uint64(len(groupInstances)),
+			Component:    componentType.String(),
+			Count:        uint64(len(groupInstances)),
+			ResourceType: resourceTypeMap[groupName],
 			Labels: []*metapb.StoreLabel{
 				{
 					Key:   groupLabelKey,
-					Value: groupLabel,
+					Value: groupName,
 				},
 			},
```

Comment:
I think we can directly place the ResourceType into Labels as we need to ensure the deploying layer should place the labels we want.